contracts/MechFixedPrice.sol

```diff
@@ -5,54 +5,19 @@ import {OlasMech} from "./OlasMech.sol";
 
 /// @title MechFixedPrice - Smart contract for OlasMech that accepts a fixed price payment for services.
 contract MechFixedPrice is OlasMech {
-    event PriceUpdated(uint256 price);
-
-    // Minimum required price
-    uint256 public price;
-
     /// @dev AgentMech constructor.
     /// @param _mechMarketplace Mech marketplace address.
     /// @param _serviceRegistry Address of the token contract.
     /// @param _serviceId Service Id.
-    /// @param _price The minimum required price.
-    constructor(address _mechMarketplace, address _serviceRegistry, uint256 _serviceId, uint256 _price)
-        OlasMech(_mechMarketplace, _serviceRegistry, _serviceId)
-    {
-        // Check for zero value
-        if (_price == 0) {
-            revert ZeroValue();
-        }
-
-        // Record the price
-        price = _price;
-    }
-
-    /// @dev Performs actions before the request is posted.
-    /// @param amount Amount of payment in wei.
-    function _preRequest(uint256 amount, uint256, bytes memory) internal virtual override {
-        // Check the request payment
-        if (amount < price) {
-            revert NotEnoughPaid(amount, price);
-        }
-    }
```

Comment:
This all becomes obsolete, as `price` becomes `maxDeliveryRate` and is now uniformal across mechs.

The checks are done on MM for now, but ultimately will be taken out to a corresponding separate escrow contract